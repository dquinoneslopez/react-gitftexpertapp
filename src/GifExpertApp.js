import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = []}) => {
    
    const [categories, setCategories] = useState(defaultCategories);
    
    // const handleAdd = () => {
    //     setCategories([...categories, 'HunterXHunter']);
    //     // setCategories(cats => [...categories, 'HunterXHunter']);
    // }
    
    return (
        <>
            <h2>GifExpertApp</h2>
            <hr/>
            
            <AddCategory setCategories={ setCategories }/>
            
            <ol>
                {
                    categories.map( category => 
                        <GifGrid key={ category } category={ category }/>
                    )
                }
            </ol>
        </>
    );
}

export default GifExpertApp;