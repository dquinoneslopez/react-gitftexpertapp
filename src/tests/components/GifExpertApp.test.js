import React from 'react'
import { shallow } from 'enzyme'
import '@testing-library/jest-dom'
import GifExpertApp from '../../GifExpertApp'

describe('Pruebas de GifExpertApp', () => {
    
    test('Debe mostrar el componente correctamente', () => {
        
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
        
    })
    
    test('Debe mostrar una lista de categorÃ­as', () => {
        
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
        
    })      
    
})
