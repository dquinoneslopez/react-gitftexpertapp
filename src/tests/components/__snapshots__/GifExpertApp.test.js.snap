// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifExpertApp Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Pruebas de GifExpertApp Debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
